### PR TITLE
Adding presto-pinot-driver module

### PR DESCRIPTION
**Pull request by xiangfu0**
## Description

Move presto-pinot-driver module from presto codebase (https://github.com/prestodb/presto-pinot-driver)

Presto repo is still on java 8, so we need to build this module with jdk8:
1. Add a new profile to set `jdk.version` for all corresponding modules
2. Add grpc client configs for presto
3. Add helper functions for presto side to avoid dependency checks from class import
 
How to build:
So when we want to build `presto-pinot-driver` module, we need to first install two JDKs, e.g. JDK 8 and JDK 11 corresponding modules with JDK 8 lib.

Add profile `presto-driver`:
```
mvn clean install -DskipTests -Ppresto-driver -T 1C -pl ':presto-pinot-driver' -am
```

## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
* [ ] Yes (Please label as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR fix a zero-downtime upgrade introduced earlier?
* [ ] Yes (Please label this as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR otherwise need attention when creating release notes? Things to consider:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed
* [ ] Yes (Please label this PR as **<code>release-notes</code>** and complete the section on Release Notes)

### CODE REVIEW

**Comment by Jackie-Jiang** [comment]
Can you please elaborate more on why we need to replicate some of the classes in this new module? And how is the module going to be released?

**Comment by xiangfu0** [comment]
> Can you please elaborate more on why we need to replicate some of the classes in this new module? And how is the module going to be released?

removed duplicated classes.

For releases, need to update the release doc of using the maven toolchain plugin

**Comment by yupeng9** [comment]
Can you explain the purpose of this change?

**Comment by xiangfu0** [comment]
> Can you explain the purpose of this change?



> Can you explain the purpose of this change?

presto is on jdk8 and pinot is on jdk 11, so we need to recompile all pinot dependencies jars for presto usage.

**Comment by yupeng9** [comment]
k. but then why there are other changes than the pom changes?

**Comment by xiangfu0** [comment]
> k. but then why there are other changes than the pom changes?

After chatted with Prestodb community, I think it's better to move presto-pinot-driver (https://github.com/prestodb/presto-pinot-driver) module in pinot codebase, then refer to this module in presto plugin directly.

Presto side has extra configs for grpc client and we expose the scatter-gather client, now we plan to own the

File: .github/workflows/pinot_tests.yml [hunk 1]
```diff
@@ -146,3 +146,19 @@ jobs:
...
 
+  presto-driver-build:
+    runs-on: ubuntu-latest
+    name: Build Presto Pinot Driver
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
```
[the remark this answers is not in the document]
Comment (xiangfu0) [reply .github/workflows/pinot_tests.yml:157]: It's possible to build java8 compatible libs by setting the `--release 8` using jdk 11. 
I will also add a jdk 8 test here.

**Review by yupeng9 — APPROVED** [review]

**Review by snleee — APPROVED** [review]
This will change our release pipeline steps when building an Apache release. It would be great if you can also update the documentation at https://cwiki.apache.org/confluence/display/PINOT/Creating+an+Apache+Release

**Comment by xiangfu0** [comment]
> This will change our release pipeline steps when building an Apache release. It would be great if you can also update the documentation at https://cwiki.apache.org/confluence/display/PINOT/Creating+an+Apache+Release

will do

**Comment by xiangfu0** [comment]
fyi: @highker

**Pull request merged**